core/DataAccess/ArchiveTableDao.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\DataAccess;
+
+use Piwik\Config;
+use Piwik\Db;
+
+/**
+ * Data Access class for querying numeric & blob archive tables.
+ */
+class ArchiveTableDao
+{
+    /**
+     * @var string
+     */
+    private $tablesPrefix;
+
+    public function __construct(Config $config)
+    {
+        $this->tablesPrefix = $config->database['tables_prefix'];
```

Comment:
Saying this as `tables_prefix` could be - in theory - changed by a plugin as there's an event for this... It's more theoretically a problem though